packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -738,13 +739,15 @@ class FormSectionComponent extends React.Component<Props> {
 
   resetDependentSelectValues = (fieldName: string) => {
     const fields = this.props.fields
-    const fieldToReset = fields.find(
+    const fieldsToReset = fields.filter(
       (field) =>
-        field.type === SELECT_WITH_DYNAMIC_OPTIONS &&
-        field.dynamicOptions.dependency === fieldName
+        (field.type === SELECT_WITH_DYNAMIC_OPTIONS || field.type === TEXT) &&
+        field?.dynamicOptions?.dependency === fieldName
     )
-    if (fieldToReset) {
-      this.props.setFieldValue(fieldToReset.name, '')
+    if (fieldsToReset.length) {
+      fieldsToReset.forEach((fieldToReset) => {
+        this.props.setFieldValue(fieldToReset.name, '')
+      })
```

Comment:
```suggestion
    fieldsToReset.forEach((fieldToReset) => {
      this.props.setFieldValue(fieldToReset.name, '')
      this.props.resetDependentSelectValues(fieldToReset.name)
    }
```
No need to check for the length of the array as `forEach` won't run if the array is empty anyway.

There was an existing bug here that only dependencies 1 level deep was checked here. So suppose that _state_ depends on _country_ and _district_ depends on _state_. And if we change _country_ then only _state_ will be reset but not _district_. But if we call `resetDependentSelectValues` recursively, all the fields on the dependency chain will be cleared.

Reply:
Agreed